org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/ClientLogHandler.java

```diff
@@ -0,0 +1,83 @@
+package org.eclipse.lsp4xml.utils;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.SimpleFormatter;
+
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4j.services.LanguageClient;
+
+/**
+ * LogHandler
+ */
+public class ClientLogHandler extends Handler{
+  private LanguageClient languageClient;
+
+  public ClientLogHandler(LanguageClient languageClient) {
+    this.languageClient = languageClient;
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if(languageClient == null) {
+      return;
+    }
+
+    String msg = formatRecord(record);
+    MessageType messageType= getMessageType(record.getLevel());
+    MessageParams mp = new MessageParams(messageType, msg);
+    languageClient.logMessage(mp);
+    
+  }
+
+  public String formatRecord(LogRecord record) {
+    StringBuilder sb = new StringBuilder();
+    sb.append(record.getMillis())
+    .append(" - ")
+    .append(record.getSourceClassName())
+    .append("#")
+    .append(record.getSourceMethodName())
+    .append(" - ")
+    .append(record.getMessage());
+    if (record.getThrown() != null) {
+      StringWriter sw = new StringWriter();
+      PrintWriter pw = new PrintWriter(sw);
+      pw.println();
+      record.getThrown().printStackTrace(pw);
+      pw.close();
+      String throwable = sw.toString();
+      sb.append(throwable);
+    }
+    return sb.toString();
+  }
+
+  private static MessageType getMessageType(Level level) {
+    MessageType mt = null;
+    String levelName = level.getName();
```

Comment:
no, check for Level.SEVERE, Level.WARNING. everything else is Level.INFO and below